tests/utils/deployment/deployment.go

```diff
@@ -235,18 +235,24 @@ func RedefineWithPriviledgedContainer(deployment *v1.Deployment) *v1.Deployment
 	return deployment
 }
 
-func RedefineWithHostPid(deployment *v1.Deployment, hostPid bool) {
+func RedefineWithHostPid(deployment *v1.Deployment, hostPid bool) *v1.Deployment {
```

Comment:
Looking at this... I still don't think we want to return the same pointer because it's already pass-by-reference.  🤔

Reply:
Fair but it is the way several other "RedefineWith..." functions in this file work. And while that's not a \*great\* reason for these functions to do it too, the non-returning way isn't consistently working so I'm thinking there may be k8s intricacies in play that I don't fully comprehend (which I suppose means my justification for this change is that I don't fully know what's going on lol)

Reply:
Also fair 😄